src/scf/self_consistent_field.jl

```diff
@@ -27,20 +29,38 @@ function next_density(ham::Hamiltonian;
      diagonalization=eigres)
 end
 
+function ScfPlotTrace(plt=plot(yaxis=:log); kwargs...)
+    energies = Float64[]
+    function callback(info)
+        if info.stage == :finalize
+            minenergy = minimum(energies[max(1, end-5):end])
+            error = abs.(energies .- minenergy)
+            error[error .== 0] .= NaN
+            extra = ifelse(:mark in keys(kwargs), (), (mark=:x, ))
+            plot!(plt, error; extra..., kwargs...)
+            display(plt)
+        else
+            push!(energies, info.energies.total)
+        end
+    end
+end
+
```

Comment:
This file is starting to get pretty busy, move all convenience callbacks/isconverged/determine_diagtol/etc to `callbacks.jl`?

Reply:
Also doc a bit this function: what does it plot?

Reply:
good point.